protocol/x/subaccounts/keeper/negative_tnc_subaccount_test.go

```diff
@@ -0,0 +1,146 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	keepertest "github.com/dydxprotocol/v4-chain/protocol/testutil/keeper"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/tracer"
+	"github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/keeper"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetSetNegativeTncSubaccountSeenAtBlock(t *testing.T) {
+	tests := map[string]struct {
+		// Setup.
+		setupTestAndPerformAssertions func(ctx sdk.Context, s keeper.Keeper)
+
+		// Expectations.
+		expectedMultiStoreWrites []string
+	}{
+		"Block height defaults to zero if not set": {
+			setupTestAndPerformAssertions: func(ctx sdk.Context, k keeper.Keeper) {
+				require.Equal(
+
+					t,
+					uint32(0),
+					k.GetNegativeTncSubaccountSeenAtBlock(ctx),
+				)
+			},
+
+			expectedMultiStoreWrites: []string{},
+		},
+		"Block height can be updated": {
+			setupTestAndPerformAssertions: func(ctx sdk.Context, k keeper.Keeper) {
+				k.SetNegativeTncSubaccountSeenAtBlock(ctx, 1)
+				require.Equal(
+
+					t,
+					uint32(1),
+					k.GetNegativeTncSubaccountSeenAtBlock(ctx),
+				)
+			},
+
+			expectedMultiStoreWrites: []string{
+				"NegSA:",
+			},
+		},
+		"Block height can be updated more than once": {
+			setupTestAndPerformAssertions: func(ctx sdk.Context, k keeper.Keeper) {
+				k.SetNegativeTncSubaccountSeenAtBlock(ctx, 1)
+				require.Equal(
+
+					t,
+					uint32(1),
+					k.GetNegativeTncSubaccountSeenAtBlock(ctx),
+				)
+
+				k.SetNegativeTncSubaccountSeenAtBlock(ctx, 2)
+				require.Equal(
+
+					t,
+					uint32(2),
+					k.GetNegativeTncSubaccountSeenAtBlock(ctx),
+				)
+
+				k.SetNegativeTncSubaccountSeenAtBlock(ctx, 3)
+				require.Equal(
+
+					t,
+					uint32(3),
+					k.GetNegativeTncSubaccountSeenAtBlock(ctx),
+				)
+
+				k.SetNegativeTncSubaccountSeenAtBlock(ctx, 10)
+				require.Equal(
+
+					t,
+					uint32(10),
+					k.GetNegativeTncSubaccountSeenAtBlock(ctx),
+				)
+			},
+
+			expectedMultiStoreWrites: []string{
+				"NegSA:",
+				"NegSA:",
+				"NegSA:",
+				"NegSA:",
+			},
+		},
+		"Block height can be updated to same block height": {
```

Comment:
for my understanding - when would we need to set this to the same height?

Reply:
I don't think this would ever be necessary, but I figured it's fine as long as it's not decreasing

Would you like me to add validation that the block height is always increasing?

Reply:
nah I think this is correct if this information eventually would come from the daemon. 

if block time > daemon loop time, then I guess this could technically happen?